### DISCUSSION TITLE
Specify WAN device used for bridge

### DISCUSSION THREAD

**Discussion by mikepaxton** [iiab/iiab#3704] [General]
Is there a way to specify which network device is used for the WAN side of the gateway bridge, ie. eth1 rather than eth0?

**Comment by holta** [comment]
Good question @mikepaxton.

This is relevant but probably doesn't fully answer your question?!

- "Exclude an Interface from br0 [LAN-side bridge]"
   https://github.com/iiab/iiab/wiki/IIAB-Networking#3-exclude-an-interface-from-br0

**Comment by tim-moody** [comment]
I don't think strictly speaking that we have a gateway bridge formed by brctl, but rather a set of firewall rules that allow or deny traffic between the LAN bridge and an upstream facing network device. I don't think we have any settings to choose the upstream device, but the firewall rules can be edited manually.

`/usr/bin/iiab-gen-iptables` manages these rules.

**Comment by tim-moody** [comment]
More info in https://github.com/iiab/iiab/wiki/IIAB-Networking#firewall-iptables

**Comment by jvonau** [comment]
> Is there a way to specify which network device is used for the WAN side of the gateway bridge, ie. eth1 rather than eth0?

@mikepaxton Before you get some bad advice (opps too late) what is the intended purpose of eth0 and what is the general layout of what you are trying to pull off? Setting the primary WAN device can simply be done within local_vars.yml by setting
`user_wan_iface: eth1`

[Ref](https://github.com/iiab/iiab/blob/master/vars/default_vars.yml#L162-L165) [Ref](https://github.com/iiab/iiab/blob/master/roles/network/tasks/computed_network.yml#L71-L75)

**Comment by tim-moody** [comment]
@jvonau thanks for the correction.

**Comment by mikepaxton** [comment]
Thanks for all the prompt replies and sorry for the delay in getting back to all of you.  Winter storms have made the power in my area spotty at best.
@jvonau What i'm trying to pull off kind of goes along with the storms we've been having.  We've needed an offline means of doing limited Internet research for some time now so, over the last few months i've been building an off-grid setup.  I stumbled upon the IIAB project and have spent the last week or so digging into it.
Our main Internet goes out fairly often as we live on top of a mountain in the middle of nowhere and or Internet provider is slow to respond.  But, we do get cellular (unless a power outage eventually takes the tower out) so, I would like to be able to USB tether my iPhone to the IIAB server for limited WAN access.  
I had limited success getting the iPhone to tether to PiOS-lite which is what i'm running IIAB on, so that might be part of the gateway interface issue.  I tried your suggested user_wan_iface: eth1, but after making the change and running iiab-network, it still clung to eth0.
However, I was playing around with the iPhone on an Ubuntu Desktop running on a Pi and found I could tether to the iPhone easily.  So, as I typing this I've got IIAB installing on Ubuntu Server and hoping for the best.  
I'm open to any suggestions on getting this to work.

**Comment by jvonau** [comment]
> Thanks for all the prompt replies and sorry for the delay in getting back to all of you. Winter storms have made the power in my area spotty at best. @jvonau What i'm trying to pull off kind of goes along with the storms we've been having. We've needed an offline means of doing limited Internet research for some time now so, over the last few months i've been building an off-grid setup. I stumbled upon the IIAB project and have spent the last week or so digging into it. Our main Internet goes out fairly often as we live on top of a mountain in the middle of nowhere and or Internet provider is slow to respond. But, we do get cellular (unless a power outage eventually takes the tower out) so, I would like to be able to USB tether my iPhone to the IIAB server for limited WAN access. I had limited success getting the iPhone to tether to PiOS-lite which is what i'm running IIAB on, so that might be part of the gateway interface issue. I tried your suggested user_wan_iface: eth1, but after making the change and running iiab-network, it still clung to eth0. However, I was playing around with the iPhone on an Ubuntu Desktop running on a Pi and found I could tether to the iPhone easily. So, as I typing this I've got IIAB installing on Ubuntu Server and hoping for the best. I'm open to any suggestions on getting this to work. 

Please be specific on what the usb dongle's device name really on the IIAB box when it's connected/tethered to the iphone...  (usb0?) do you intend to have the LAN clients access the internet via the iphone or just the IIAB box by itself?

**Comment by mikepaxton** [comment]
When the iPhone tethers on Ubuntu it uses the device name of enxdeb54f532531 which is enx + MAC address of the iPhone.  It tethers as type ethernet.  
At this time I think all clients will strictly be connecting wirelessly, most LAN devices would be down due to power outages.  
As a side note, I installed Ubuntu Server 22.04 on my Pi in hopes of duplicating my success with connecting the iPhone to Pi's running Ubuntu Desktop 22.04.  So far i've not found the magic bullet that gets this whole setup working on a server.  It's has to be a driver that gets installed by default on Ubuntu Desktop and not on the server.   Technically speaking, it's supposed to be these three drivers which make the magic happen:  usbmuxd libimobiledevice6 libimobiledevice-utils

Edit:  I suspect the tethering issue is a dhcp issue as i get a constant (getting IP configuration) message while tethering and after several minutes it times out.

**Comment by jvonau** [comment]
I have had success with using android based devices for upstream access via usb with an Ubuntu desktop, but I can't recall if the server release behaved the same way. Using wifi for the upstream should work out of the box on IIAB but setting up the upstream wifi connection on server can be a PITA. Think you are on the right track with with iOS [libraries](https://github.com/libimobiledevice/libimobiledevice/issues/1019) being the magic sauce. [Ref](https://askubuntu.com/questions/1461639/how-do-i-set-up-internet-usb-tethering-with-an-iphone-in-ubuntu-22-04) mentions 'ipheth-utils' and enabling 'systemctl start usbmuxd', might have to reboot as NetworkManager is not active but could be if you mess with netplan to switch the networking backend that is in use. Good Luck 

Edit:
If the LAN clients need to access the internet you would need to run `iiab-network` or edit /etc/iiab/iiab.env changing IIAB_WAN_DEVICE= to be your current gateway device and running `sudo iiab-gen-iptables` or `sudo iiab-internet-on`

**Comment by mikepaxton** [comment]
@jvonau For the time being i've shelved the iPhone USB tethering.  I want to focus on getting the rest of the iiab-server installed and running with content.  
I had no luck with Ubuntu Server and the tethering.  From what I've read, libimobiledevice6 and it's accompanying packages are geared more towards adding developers additional Linux tools for iOS software development.  If memory serves me correctly, ipheth-utils isn't even listed on the libimobledevice developers website as a needed library.  Whereas ipheth-utils seems to be the go to library for tech blogger iPhone pairing instruction sites.  
Hmmmmm.....just had a thought, I wonder if they might be interfering with each other?  Don't recall if I tried ipheth-utils on its own, I'll have to play around with that.
For now, i'm trying to chase down an init-failed error on the Admin-Console.  Oddly, it was working fine when I stepped away from my computers to get some dinner and a bit of rest.  Now that it's 1am in the morning it doesn't seem to want to co-operate with me.  But, if need be I'll leave that trouble shooting on another discussion thread.  Don't want to pollute this thread incase someone else down the road stumbles across an iPhone tethering issue.
